### PR TITLE
refactor(provider/kubernetes): Remove SERVER_GROUP label dependency when loading instance data and f…

### PR DESCRIPTION
**Pull request by chlung**
…ix unit test..

We prefer small, well tested pull requests.

Please refer to [Contributing to Spinnaker](http://www.spinnaker.io/docs/contributing-to-spinnaker).

When filling out a pull request, please consider the following:

* Follow the commit message conventions [found here](http://www.spinnaker.io/v1.0/docs/how-to-submit-a-patch).
* Provide a descriptive summary for your changes.
* If it fixes a bug or resolves a feature request, be sure to link to that issue.
* Add inline code comments to changes that might not be obvious.
* Squash your commits as you keep adding changes.
* Add a comment to @spinnaker/reviewers for review if your issue has been outstanding for more than 3 days.

Note that we are unlikely to accept pull requests that add features without prior discussion. The best way to propose a feature is to open an issue first and discuss your ideas there before implementing them.

### CODE REVIEW

**Comment by chlung** [comment]
@lwander Please review this change which is intending to remove hard coded label dependency.  Thanks.

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/model/KubernetesInstance.groovy [hunk 1]
```diff
@@ -74,8 +75,10 @@ class KubernetesInstance implements Instance, Serializable {
 
     this.health << (Map<String, String>) mapper.convertValue(new KubernetesHealth(pod), new TypeReference<Map<String, String>>() {})
 
-    this.controllerName = pod.metadata?.labels?.get(KubernetesUtil.SERVER_GROUP_LABEL) ?:
-        pod.metadata?.labels?.get(KubernetesUtil.JOB_LABEL) ?: null
+    if (pod.metadata?.ownerReferences) {
+      this.controllerName = pod.metadata?.ownerReferences.get(0).getName()
```
Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/model/KubernetesInstance.groovy]: Can you make the `.getName()` null safe as well? e.g. `?.getName()`

**Comment by chlung** [comment]
@lwander Updated per your feedback.  Please review.  Thanks.

**Comment by chlung** [comment]
@lwander Updated per your feedback.  Please review.  Thanks.

**Comment by lwander** [comment]
Please rebase & I'll merge

**Comment by chlung** [comment]
@lwander rebase this branch.  Please merge.  Thanks.

**Comment by lwander** [comment]
This isn't up-to-date, you need to rebase first. Also, please squash into 1 commit that follows our naming conventions: https://spinnaker.github.io/community/contributing/submitting/#commit-message-conventions

**Pull request merged**